File: filter-pattern/js/src/CriteriaSingle.js
import { Criteria } from './Criteria.js'

// 定义子类覆写父类抽象方法
export class CriteriaSingle extends Criteria {
  filter(persons) {
    const singlePersons = []
    for (const person of persons) {
      if (person.getStatus().toUpperCase() === 'SINGLE') {
        singlePersons.push(person)
      }
    }
    return singlePersons
  }
}

File: filter-pattern/java/src/CriteriaSingle.java
package src;

import java.util.*;

// 根据标准接口实现按属性的过滤
public class CriteriaSingle implements Criteria {

   @Override
   public List<Person> filter(List<Person> persons) {
      List<Person> singlePersons = new ArrayList<Person>();
      for (Person person : persons) {
         if (person.getStatus().equalsIgnoreCase("SINGLE")) {
            singlePersons.add(person);
         }
      }
      return singlePersons;
   }
}